tests/tensorflow2/test_keras.py

```diff
@@ -0,0 +1,240 @@
+`python tests/tensorflow2/test_keras.py` from the main directory.
+"""
+
+# Third Party
+import pytest
+import tensorflow.compat.v2 as tf
+from tests.tensorflow.utils import create_trial_fast_refresh
+
+# First Party
+import smdebug.tensorflow as smd
+from smdebug.core.access_layer import has_training_ended
+from smdebug.core.collection import CollectionKeys
+from smdebug.core.reduction_config import ALLOWED_NORMS, ALLOWED_REDUCTIONS
+from smdebug.exceptions import TensorUnavailableForStep
+from smdebug.tensorflow import ReductionConfig, SaveConfig
+
+
+def helper_keras_fit(
+    trial_dir,
+    save_all=False,
+    include_collections=None,
+    reduction_config=None,
+    save_config=None,
+    hook=None,
+    eager=True,
+    steps=None,
+    add_callbacks=None,
+):
+    if not eager:
+        tf.compat.v1.disable_eager_execution()
+
+    mnist = tf.keras.datasets.mnist
+    (x_train, y_train), (x_test, y_test) = mnist.load_data()
+    x_train, x_test = x_train / 255, x_test / 255
+
+    model = tf.keras.models.Sequential(
+        [
+            tf.keras.layers.Flatten(input_shape=(28, 28)),
+            tf.keras.layers.Dense(128, activation="relu"),
+            tf.keras.layers.Dropout(0.2),
+            tf.keras.layers.Dense(10, activation="softmax"),
+        ]
+    )
+
+    if hook is None:
+        if save_config is None:
+            save_config = SaveConfig(save_interval=3)
+
+        hook = smd.KerasHook(
+            trial_dir,
+            save_config=save_config,
+            save_all=save_all,
+            include_collections=include_collections,
+            reduction_config=reduction_config,
+        )
+
+        if not save_all and include_collections is not None:
+            for cname in hook.include_collections:
+                if cname not in include_collections:
+                    hook.get_collection(cname).save_config = SaveConfig(end_step=0)
+
+    opt = tf.keras.optimizers.Adam()
+
+    opt = hook.wrap_optimizer(opt)
+    model.compile(optimizer=opt, loss="sparse_categorical_crossentropy", metrics=["accuracy"])
+    hooks = []
+    if add_callbacks:
+        if "tensorboard" in add_callbacks:
+            hooks.append(
+                tf.keras.callbacks.TensorBoard(
+                    log_dir="/tmp/logs", histogram_freq=1, write_grads=True, write_images=True
+                )
+            )
+    hooks.append(hook)
+
+    if steps is None:
+        steps = ["train"]
+    for step in steps:
+        if step == "train":
+            model.fit(x_train, y_train, epochs=1, steps_per_epoch=10, callbacks=hooks, verbose=0)
+        elif step == "eval":
+            model.evaluate(x_test, y_test, steps=10, callbacks=hooks, verbose=0)
+        elif step == "predict":
+            model.predict(x_test[:100], callbacks=hooks, verbose=0)
+
+    hook.close()
+
+
+@pytest.mark.slow
+@pytest.mark.parametrize("saveall", [True, False])
+def test_keras_fit(out_dir, tf_eager_mode, saveall):
```

Comment:
where is out_dir and tf_eager_mode coming from?

Reply:
conftest.py (tests/conftest.py)

Reply:
i dont see tf_eager_mode in above file.

Reply:
https://github.com/awslabs/sagemaker-debugger/pull/150/files#diff-484462fced51d1a06b1d93b4a44dd535R49

Reply:
introduced in this PR